packages/core/buidler.config.js

```diff
@@ -24,9 +41,145 @@ task("test")
+        console.log(
+          chalkPipe("orange.bold")(
+            `Unable to find constructor args for ${contractName} at ${address}, skipping verification...`
+          )
+        );
+        continue;
+      }
+
+      // Get the saved constructor params
+      const constructorArguments = deployedArgs[address];
+
+      // ABI encode constructor params
+      const abi = (await readArtifact(bre.config.paths.artifacts, contractName)).abi;
+      const constructorAbi = abi.filter(x => x.type === "constructor")[0];
+      let abiEncodedConstructorArgs = "";
+      if (constructorAbi) {
+        // Using ethers here because web3.js doesn't support custom tuple construction (e.g. FixedPoint)
+        // https://github.com/ethereum/web3.js/issues/1241
+        abiEncodedConstructorArgs = ethers.utils.defaultAbiCoder.encode(constructorAbi.inputs, constructorArguments);
+
+        // Remove '0x'
+        abiEncodedConstructorArgs = abiEncodedConstructorArgs.slice(2);
+      }
+
+      // Format contract name according to etherscan
+      // more info @ https://etherscan.io/apis#contracts
+      const constractPath = Object.keys(solcInput.sources).filter(x => x.includes(`/${contractName}.sol`))[0];
+      const etherscanContractName = `${constractPath}:${contractName}`;
+
+      // Request on etherscan
+      // JSON format - https://etherscan.io/apis#contracts
+      const verificationRequest = {
+        apikey: etherscan.apiKey,
+        module: "contract",
+        action: "verifysourcecode",
+        contractaddress: address,
+        sourceCode: JSON.stringify(solcInput),
+        contractname: `${etherscanContractName}`,
+        codeformat: "solidity-standard-json-input",
+        compilerversion: solcFullVersion,
+        constructorArguements: abiEncodedConstructorArgs
+      };
+
+      try {
+        console.log(chalkPipe("yellow.bold")(`Attempting to verify ${contractName} at ${address}`));
+        const response = await verifyContract(etherscan.url, verificationRequest);
+        console.log(
+          `Submitted ${contractName} contract at ${address} for verification on etherscan (GUID: ${response.message}). Waiting for verification result...`
+        );
+        await getVerificationStatus(etherscan.url, response.message);
+        console.log(chalkPipe("green.bold")(`Successfully verified ${contractName} at ${address} on etherscan`));
+      } catch (e) {
+        if (e.toString().includes("already verified")) {
+          console.log(chalkPipe("orange.bold")(`${contractName} at ${address} is already verified on etherscan`));
+        } else {
+          console.log(
+            chalkPipe("red.bold")(
+              `Failed to verified ${contractName} at ${address} on etherscan, reason: ${e.toString()}`
+            )
+          );
+        }
+      }
+    }
+  });
+
 module.exports = {
   solc: {
-    version: "0.6.12",
+    version: solcVersion,
     optimizer: {
       enabled: true,
       runs: 199
```

Comment:
Should you change the `networks` config to the user's connected network?

Reply:
This networks is a config when buidler runs tests, e.g. `npx buidler test --network localhost/buidlerevm`.

What do you mean by "user connected network"?

Reply:
what if I want to use Kovan or Mainnet network would it work for that? Sorry I don't have much familiarity with `buidler`

Reply:
If you'd like to run tests for the kovan network, you'll need to add the network config to the "networks" section. This PR is about a separate topic - verifying contracts, please keep it at that.

To verify the contracts, you need to files to be saved in the truffle-registry (i.e. in `networks/` directory). So for example, if you've deployed the kovan contracts, it will appear as `networks/42.json`. This PR also captures the constructor args for each deployed contract, and saves them into `networks/42_args.json`.

To verify the deployed contracts for kovan you'll run `npx buidler verify-etherscan --network-id 42`. It'll try and find the files `42.json` and `42_args.json` in the `network` directory, and combined them with `solc-input.json` before uploading them to etherscan.

Reply:
> If you'd like to run tests for the kovan network, you'll need to add the network config to the "networks" section. This PR is about a separate topic - verifying contracts, please keep it at that.

Sorry I think i am confused. You're saying that the `networks` object has no impact on verification? Only affects the tests?

Reply:
Yeap! You supply the `network-id` to the cli here, https://github.com/UMAprotocol/protocol/blob/feat/etherscan-automate-verification/packages/core/buidler.config.js#L48

Reply:
Just to be clear, I _think_ what you're saying @kendricktan is that the verification does not require buidler to have any non-test networks set up (like we have for truffle). The verification task is independent of (and doesn't rely on) the declared networks in the buidler config. All it needs is to be able to compile (network agnostic) and see the `networks/` files.

SGTM!